source/adapters/level_zero/event.cpp

```diff
@@ -434,8 +456,22 @@ ur_result_t urEnqueueEventsWaitWithBarrierExt(
         *OutEvent ///< [in,out][optional] return an event object that identifies
                   ///< this particular command instance.
 ) {
-  return ur::level_zero::urEnqueueEventsWaitWithBarrier(
-      Queue, NumEventsInWaitList, EventWaitList, OutEvent);
+  bool InterruptBased =
+      EnqueueExtProp &&
+      (EnqueueExtProp->flags & UR_EXP_ENQUEUE_EXT_FLAG_LOW_POWER_EVENTS);
+  if (InterruptBased) {
```

Comment:
I'm not sure I understand the intention here or why we need to cast the function.

Reply:
-We make sure that EnqueueExtProp is not a nullptr then check the flag to see if we want to use LOW_POWER.
-EnqueueEventsWaitWithBarrier(ur_queue_handle_t_*&, uint32_t&, ur_event_handle_t_* const*&, ur_event_handle_t_**&, bool is overloaded therefore we needed to specify.

Reply:
I think this would be cleaner if you move the implementation of urEnqueueEventsWaitWithBarrier() to urEnqueueEventsWaitWithBarrierExt() and have urEnqueueEventsWaitWithBarrier call urEnqueueEventsWaitWithBarrierExt with the check for the queue property for interrupt based.

Reply:
Good idea, let me add that in. Thanks!